### PR TITLE
TSK-425: Supported team settings

### PR DESCRIPTION
**Pull request by mrsadman99**
Signed-off-by: Anton Brechka [anton.brechka@xored.com](mailto:anton.brechka@xored.com)

# Contribution checklist

## Checklist

* [ ] - UI test added to added/changed functionality?
* [ ] - Screenshot is added to PR if applicable ?
* [ ] - Does the code work? Check whether function and logic are correct.
* [ ] - Does Changelog.md is updated with changes?
* [ ] - Does the translations are up to date?
* [ ] - Does it well tested?
* [ ] - Tested for Chrome.
* [ ] - Tested for Safari.
* [ ] - Go through the changed code looking for typos, TODOs, commented LOCs, debugging pieces of code, etc.
* [ ] - Rebase your branch onto master and upstream branch
* [ ] - Is there any redundant or duplicate code?
* [ ] - Are required links are linked to PR?
* [ ] - Does new code is well documented ?

## Related issues

A list of closed updated issues

### CODE REVIEW

File: plugins/tracker-resources/src/components/CreateIssue.svelte [hunk 1]
```diff
@@ -113,51 +138,19 @@
...
-      dueDate: null,
-      comments: 0,
-      subIssues: 0,
-      parents: [],
-      reportedTime: 0,
-      estimation: 0,
-      reports: 0,
-      childInfo: []
-    }
+    object = defaultIssue
```
Comment (haiodo) [inline plugins/tracker-resources/src/components/CreateIssue.svelte]:
```suggestion
    object = {...defaultIssue}
```

Over wise create next one will use previous values.

Comment (mrsadman99) [reply plugins/tracker-resources/src/components/CreateIssue.svelte]: Done.

**Comment by haiodo** [comment]
Please change $in

image

**Comment by mrsadman99** [comment]
> Please change $in
> 
> image

Done.

File: models/tracker/src/migration.ts [hunk 2]
```diff
@@ -178,6 +203,23 @@ async function upgradeIssueStatuses (tx: TxOperations): Promise<void> {
   }
 }
 
+async function upgradeIssueTimeReportSettings (tx: TxOperations): Promise<void> {
+  const issues = await tx.findAll(tracker.class.Issue, {
+    defaultTimeReportDay: { $exists: false },
+    workDayLength: { $exists: false }
+  })
+
+  const teams = await tx.findAll(tracker.class.Team, { $in: Array.from(new Set(issues.map((issue) => issue.space))) })
```
Comment (haiodo) [inline models/tracker/src/migration.ts]: Top level $in is not supported, please use 
```suggestion
  const teams = await tx.findAll(tracker.class.Team, { _id: { $in: Array.from(new Set(issues.map((issue) => issue.space))) }})
```

Comment (mrsadman99) [reply models/tracker/src/migration.ts]: Done.

**Review by haiodo — APPROVED** [review]

**Pull request merged**